lib/ansible/module_utils/facts/system/os_release.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import shlex
+import os
+
+from ansible.module_utils.facts.utils import get_file_content
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+
+
+class OSReleaseFactCollector(BaseFactCollector):
+    name = 'os-release'
+    _fact_ids = set(['os_release'])
+
+    def parse_file(self, path):
+        result = {}
+        for line in get_file_content(path).splitlines():
+            if line.startswith('#'):
+                continue
+            for t in shlex.split(line):
```

Comment:
One odd thing that has come up with the os-release parsing code in module_utils/facts/system/distribution.py is the quoting of some of the right hand side of lines. 

I'll track down the bug momentarily, but at least one distro did odd things about what was/wasn't quoted  (either VERSION wasnt quoted or ID was quoted?). The lsb/os-release spec is kind of vague on that, and different '/bin/lsb_release' tools seemed to do different things.

This isn't something that has to be addressed since I don't think there was a clear correct answer, but just mentioning it for context.

Reply:
So, while lsb is vague on that, I think, but the os-release specs is clear:
"Variable assignment values must be enclosed in double or single quotes if they include spaces, semicolons or other special characters outside of A–Z, a–z, 0–9. Shell special characters ("$", quotes, backslash, backtick) must be escaped with backslashes, following shell style."

Now of course, being on the right side of the specs do not help users :/